lib/util.js

```diff
@@ -658,4 +659,38 @@ exports.getCompareAndLogEnvAndAgentProxies = function (agentOptions) {
 */
 exports.removeScheme = function (input) {
   return input.toString().replace(/(^\w+:|^)\/\//, '');
+};
+
+/**
+ * Checks if the provided file or directory permissions are correct.
+ * @param filePath
+ * @param expectedMode
+ * @param fsPromisesModule
+ * @returns {Promise<boolean>} resolves always to true for Windows
+ */
+exports.isFileModeCorrect = async function (filePath, expectedMode, fsPromisesModule) {
+  if (os.platform() === 'win32') {
+    return true;
+  }
+  const fsPromises = typeof fsPromisesModule !== 'undefined' ? fsPromisesModule : require('fs/promises');
+  return await fsPromises.stat(filePath).then((stats) => {
+    // we have to limit the number of LSB bits to 9 with the mask, as the stats.mode starts with "40"
```

Comment:
can we describe why 40 is so important?

Reply:
Added more description why we are limiting the result to 9 bits. The rest of the mode (MSBs) are describing the type of the `file`, e.g. `4` informs that it is a directory.